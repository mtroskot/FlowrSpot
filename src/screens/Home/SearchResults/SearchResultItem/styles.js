import { StyleSheet } from 'react-native';
import { dimensions, fonts } from 'src/styles';
const { rem } = dimensions;
const imageSize = 50 * rem;

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    backgroundColor: '#fff',
    padding: 5 * rem
  },
  avatarView: {
    flex: 0.2,
    paddingLeft: 20 * rem
  },
  image: {
    height: imageSize,
    width: imageSize,
    borderRadius: imageSize / 2,
    resizeMode: 'cover'
  },
  contentView: {
    justifyContent: 'center',
    flex: 0.8
  },
  mainContentText: {
    ...fonts.style.normal,
    fontSize: 16 * rem,
    color: '#000'
  },
  contentText: {
    ...fonts.style.italic,
    fontSize: 12 * rem,
    marginTop: 5 * rem,
    color: '#666666'
  }
});

export default styles;
