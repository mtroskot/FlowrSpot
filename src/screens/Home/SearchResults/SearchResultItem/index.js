import React from 'react';
import { Image, Text, TouchableOpacity, View } from 'react-native';
import PropTypes from 'prop-types';
import styles from 'src/screens/Home/SearchResults/SearchResultItem/styles';

const SearchResultItem = props => {
  const { id, name, latin_name, profile_picture, onResultPress } = props;
  return (
    <TouchableOpacity activeOpacity={0.95} onPress={() => onResultPress(id)} style={styles.container}>
      <View style={styles.avatarView}>
        <Image source={{ uri: `https:${profile_picture}` }} style={styles.image} />
      </View>
      <View style={styles.contentView}>
        <Text style={styles.mainContentText}>{name}</Text>
        <Text style={styles.contentText}>{latin_name}</Text>
      </View>
    </TouchableOpacity>
  );
};

SearchResultItem.propTypes = {
  id: PropTypes.number.isRequired,
  name: PropTypes.string.isRequired,
  latin_name: PropTypes.string.isRequired,
  profile_picture: PropTypes.string.isRequired,
  onResultPress: PropTypes.func.isRequired
};

export default React.memo(SearchResultItem);
