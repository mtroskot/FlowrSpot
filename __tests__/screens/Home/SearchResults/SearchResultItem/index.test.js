import React from 'react';
import { TouchableOpacity } from 'react-native';
import { shallow } from 'enzyme';
import SearchResultItem from 'src/screens/Home/SearchResults/SearchResultItem';

describe('SearchResultItem wrapper', () => {
  it('renders correctly', () => {
    const props = {
      id: 'id',
      name: 'name',
      latin_name: 'latin_name',
      profile_picture: 'profile_picture',
      onResultPress: jest.fn()
    };

    const wrapper = shallow(<SearchResultItem {...props} />);
    expect(wrapper.first().type()).toEqual(TouchableOpacity);
    expect(wrapper.find('TouchableOpacity')).toHaveLength(1);
    expect(wrapper.find('View')).toHaveLength(2);
    expect(wrapper.find('Image')).toHaveLength(1);
    expect(
      wrapper
        .find('Image')
        .at(0)
        .prop('source').uri
    ).toEqual(`https:${props.profile_picture}`);
    expect(wrapper.find('Text')).toHaveLength(2);
    expect(
      wrapper
        .find('Text')
        .at(0)
        .props().children
    ).toEqual(props.name);
    expect(
      wrapper
        .find('Text')
        .at(1)
        .props().children
    ).toEqual(props.latin_name);
    expect(wrapper).toMatchSnapshot();
  });
});

describe('interaction', () => {
  const props = {
    id: 'id',
    name: 'name',
    latin_name: 'latin_name',
    profile_picture: 'profile_picture',
    onResultPress: jest.fn()
  };

  const wrapper = shallow(<SearchResultItem {...props} />);
  it('pressing TouchableOpacity, should call the onPress callback', () => {
    wrapper
      .find('TouchableOpacity')
      .first()
      .prop('onPress')();
    expect(props.onResultPress).toHaveBeenCalledTimes(1);
  });
});
