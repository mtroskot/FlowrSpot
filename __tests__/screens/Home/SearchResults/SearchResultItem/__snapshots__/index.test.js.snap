// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResultItem wrapper renders correctly 1`] = `
<TouchableOpacity
  activeOpacity={0.95}
  onPress={[Function]}
  style={
    Object {
      "backgroundColor": "#fff",
      "flexDirection": "row",
      "padding": 9.868421052631579,
    }
  }
>
  <View
    style={
      Object {
        "flex": 0.2,
        "paddingLeft": 39.473684210526315,
      }
    }
  >
    <Image
      source={
        Object {
          "uri": "https:profile_picture",
        }
      }
      style={
        Object {
          "borderRadius": 49.34210526315789,
          "height": 98.68421052631578,
          "resizeMode": "cover",
          "width": 98.68421052631578,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "flex": 0.8,
        "justifyContent": "center",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#000",
          "fontFamily": "Ubuntu-Regular",
          "fontSize": 31.57894736842105,
        }
      }
    >
      name
    </Text>
    <Text
      style={
        Object {
          "color": "#666666",
          "fontFamily": "Ubuntu-Italic",
          "fontSize": 23.684210526315788,
          "marginTop": 9.868421052631579,
        }
      }
    >
      latin_name
    </Text>
  </View>
</TouchableOpacity>
`;
